cmd/kubeadm/app/cmd/init_test.go

```diff
@@ -81,7 +81,11 @@ func TestNewInitData(t *testing.T) {
 	}{
 		// Init data passed using flags
 		{
-			name: "pass without any flag (use defaults)",
+			name: "pass without any flag except the cri socket (use defaults)",
+			// set the cri socket here, otherwise it would fail if the testcase is run on the node with multiple cri endpoints configured
+			flags: map[string]string{
+				options.NodeCRISocket: constants.UnknownCRISocket,
+			},
 		},
```

Comment:
The below test cases (`expectError: true`) may fail for multi cri when there are. Not fail for the expected reason.

Should we update them as well?

Reply:
```
cmd.Flags().Set(options.NodeCRISocket: constants.UnknownCRISocket)
```
I think adding the above set before the below code is another way to fix this.

https://github.com/kubernetes/kubernetes/blob/cbc9db31a27abc4dd672915abe888ded1e4f1a43/cmd/kubeadm/app/cmd/init_test.go#L199-L202

Reply:
> The below test cases (`expectError: true`) may fail for multi cri when there are. Not fail for the expected reason.
> 
> Should we update them as well?

yes, some of them need to be updated as well.

Reply:
> ```
> cmd.Flags().Set(options.NodeCRISocket: constants.UnknownCRISocket)
> ```
> 
> I think adding the above set before the below code is another way to fix this.
> 
> https://github.com/kubernetes/kubernetes/blob/cbc9db31a27abc4dd672915abe888ded1e4f1a43/cmd/kubeadm/app/cmd/init_test.go#L199-L202

hmm... this sound more neat.